### PR TITLE
Add OpenShift integration tests

### LINKED ISSUE #349: Run integration tests against OpenShift cluster as part of CI
**Description**

In order to guarantee changes to the charts or the images deployed do not break OpenShift deployments, we need to test every PR/change request. This gives us confidence on every change that a new release will not break any existing OpenShift deployments due to OpenShift incompatibility.

**Acceptance criteria**

For every PR, we should test the following:

- [x] Deploy the char to an OpenShift cluster in its own project
- [x] Deploy a process instance to Zeebe, create an instance and complete it
- [x] Create new users in Identity
- [x] Login into Operate using the created user
- [x] View created instances in Operate (possibly enough to use the headless API?)
- [x] Cancel a process instance from Operate
- [x] Login to Tasklist using the created user
- [x] View a user task in Tasklist (enough to use the GraphQL API?)
- [x] Complete a user task in Tasklist (enough to use the GraphQL API?)
- [x] Login to Optimize using the created user
- [x] View events from Zeebe in Optimize

**Additional notes**

While most of these can be tested using some headless APIs, some of them cannot (e.g. anything starting with view). I'm not 100% sure if it's worth spending the time to test these, and will coordinate with the other teams to figure out what is a minimal smoke test to guarantee everything works. Perhaps the health check are sufficient? We still do want to test the interaction _between_ the services, though.

### PR DESCRIPTION
**Pull request by npepinpe**
**Description**

This PR adds integration tests targeting an existing OpenShift cluster. The tests expect the cluster to exist _and_ that a service account exists, with the following permissions:

- create `projectrequests`
- get, list, and watch `nodes`

Additionally, two secrets must exist in the repository:

- `OPENSHIFT_SERVER_URL`, which points to your cluster's API login server
- `OPENSHIFT_TOKEN`, which is your user or service account's token.

See #348 on how to set up the service account.

Because OpenShift works with `projects` and not `namespaces`, I split the definition of the `integrationSuite` type into two files. The file `integration_test.go` now is meant for integration tests that do not target OpenShift, and the `openshift_test.go` houses the OpenShift specific tests (only a simple end to end one). All the other building blocks with which we write tests are pushed to `integration_suite.go`.

The OpenShift tests will run only with the build tags `integration` and `openshift`, and the normal integration tests will run with `integration`, but will **not** run if the `openshift` tag is specified. You can use the `make it-os` target to run the OpenShift integration tests locally.

The integration tests expect the current k8s context to already point to an OpenShift cluster, using an account as specified above. They will then create a project as a setup, and delete it on teardown.

Because the tests use the bugged version of Helm >= 3.2.x, we need to use the post-renderer as described in the README.md in the `/openshift` folder. It's terribly hacky - our test is Go binary which launches a Helm process which launches a bash script which uses sed :clown_face: - but it works. Happy to hear suggestions!

The workflow will run the `it-os` target, but before installs the right `helm`, `kustomize`, and `oc` versions, logs in, verify it has the right permissions, etc. It's already set up to support a matrix of multiple versions - we will simply need multiple secrets for each version.

**Related issues**
closes #349

### CODE REVIEW

**Comment by npepinpe** [comment]
This PR is blocked by #362

**Comment by npepinpe** [comment]
@aabouzaid Same as the other PR, do as little or as much as you want :+1: Your previous review was very useful, but don't worry if you're busy about blocking this.

**Comment by Zelldon** [comment]
Sorry @npepinpe I will try to review it today

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -143,6 +155,22 @@ _Conjunction:_
 
 All variables and the corresponding documentation are reflected in the [README](https://github.com/camunda/camunda-platform-helm/blob/main/charts/camunda-platform/README.md). Please make sure to update the README as well, if changing or adding new variables. Their exist an helper script to generate a markdown like structure based on the `.yaml` file documentation. You can find it [here](https://github.com/camunda/camunda-platform-helm/blob/main/charts/camunda-platform/convertValuesDoc.sh).
 
+
+## CI
+
+CI is performed via GitHub actions. You can find the workflows [here](/.github/workflows). Currently, CI consists of:
+
+- [a set of Helm char tests](/.github/workflows/charttesting.yml)
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply CONTRIBUTING.md:163]: I thought it actually ran `helm test` :see_no_evil:

File: Makefile [hunk 2]
```diff
@@ -8,11 +8,15 @@ releaseName=camunda-platform-test
 test:	deps
 	go test ./...
 
-# it: runs the integration tests agains the current kube context
+# it: runs the integration tests against the current kube context
 .PHONY: it
 it:	deps
 	go test -p 1 -timeout 1h -tags integration ./.../integration
 
+# it-os: runs a subset of the integration tests against the current Openshift cluster
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply Makefile:16]: Oh yeah, oops, I forgot :see_no_evil:

File: charts/camunda-platform/test/integration/integration_suite.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	s.Require().EqualValues(3, topology.ReplicationFactor)
+}
+
+func (s *integrationSuite) resolveKeycloakServiceName() string {
+	// Keycloak truncates at 20 chars since the node identifier in WildFly is limited to 23 characters.
+	// see https://github.com/bitnami/charts/blob/master/bitnami/keycloak/templates/_helpers.tpl#L2
+	keycloakServiceName := fmt.Sprintf("%s-keycl", s.release)
+	keycloakServiceName = strings.TrimSuffix(keycloakServiceName[:20], "-")
+	return keycloakServiceName
+}
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply charts/camunda-platform/test/integration/integration_suite.go:274]: Yeah, it's everything that was in `integration_test.go` but that wasn't a test itself :+1:

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Zelldon) [reply CONTRIBUTING.md:163]: I think `ct`
 does, but I think we haven't really something specified for it :man_shrugging:

[the remark this answers is not in the document]
Comment (npepinpe) [reply CONTRIBUTING.md:163]: Ah, it seems it doesn't, we only run `ct lint` and not `ct lint-and-install`, so I guess it won't. TIL :+1:

**Pull request merged**